torch/testing/_internal/common_utils.py

```diff
@@ -940,6 +940,35 @@ def get_comparison_dtype(a, b):
 
     return compare_dtype
 
+def compare_with_reference(torch_fn, ref_fn, sample_input):
+    # Compares torch function with reference function for given sample input (object of SampleInput)
+    # Note: only values are compares, type comparison is not done here
+    try:
+        n_inp, n_args, _ = sample_input.numpy()
+    except TypeError:
+        # TODO: This needs a comment
+        return
+    t_inp, t_args, _ = sample_input.splat()
+
+    t_kwargs, n_kwargs = torch_fn.sample_kwargs(t_inp.device, None, t_inp)
+    actual = torch_fn(t_inp, *t_args, **t_kwargs)
+    expected = ref_fn(n_inp, *n_args, **n_kwargs)
+
+    # Crafts a custom error message for smaller, printable tensors
+    if t_inp.numel() < 10:
+        msg = ("Failed to produce expected results! Input tensor was"
+               " {0}, torch result is {1}, and reference result is"
+               " {2}.").format(t_inp, actual, expected)
+    else:
+        msg = None
+
+    if isinstance(actual, torch.Tensor):
+        torch.testing.assert_close(actual, torch.as_tensor(expected, dtype=actual.dtype), msg=msg, check_device=False)
+    else:
+        for x, y in zip(expected, actual):
+            # Testing multi-outputs results
+            torch.testing.assert_close(actual, torch.as_tensor(expected, dtype=actual.dtype), msg = msg, check_device=False)
```

Comment:
`torch.testing.assert_close` has container support so there is no need to for this looping here. Plus, this is missing a check for the same amount of elements, since `zip` stops whenever one iterable is exhausted. So for example

```python
for x, y in zip([], [1, 2, 3]):
    assert False
```

passes just fine although it shouldn't.

Reply:
> torch.testing.assert_close has container support so there is no need to for this looping here

Cool, thanks for the info @pmeier! Will remove the looping.

> Plus, this is missing a check for the same amount of elements, since zip stops whenever one iterable is exhausted

Agreed, good point. I'll add an assertion here.

Reply:
use assertEqual and make this a method on the testcase so it can inherit atol (and, in the future, rtol) from decorators like precisionOverride

Reply:
(assertEqual also works on lists so I think you can just call it on the output)

Reply:
In the future `self.assertEqual` will use `torch.testing.assert_close` internally, so we get all features from it.